client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -523,33 +529,37 @@ void main(void)
 
 		// Sky reflection
 		col.rgb += reflection_color * pow(fresnel_factor, 2.0) * 0.5 * brightness_factor;
-		vec3 water_reflect_color = 12.0 * dayLight * fresnel_factor * mtsmoothstep(0.85, 0.9, pow(clamp(dot(reflect_ray, viewVec), 0.0, 1.0), 32.0)) * max(1.0 - shadow_uncorrected, 0.0);
 
-		// This line exists to prevent ridiculously bright reflection colors.
-		water_reflect_color /= clamp(max(water_reflect_color.r, max(water_reflect_color.g, water_reflect_color.b)) * 0.375, 1.0, 400.0);
-		col.rgb += water_reflect_color * f_adj_shadow_strength * brightness_factor;
+		vec3 water_reflect_color =
+			2.0 * specular_intensity * sunTint * dayLight * fresnel_factor * max(1.0 - shadow_uncorrected, 0.0) *
+			mtsmoothstep(0.85, 0.9, pow(clamp(dot(reflect_ray, viewVec), 0.0, 1.0), 32.0));
+
+		// Sun reflection
+		col.rgb += water_reflect_color * f_shadow_factor * brightness_factor;
```

Comment:
This seems to make the water reflections much brighter.
Can you explain f_shadow_factor vs. adj_shadow_shadow_strength?

Reply:
Doesn't actually make them brighter for me, at least as far as I can tell...
`f_shadow_factor` essentially just ensures the reflections vanish during sunrise and sunset and get dimmer at night, whereas `f_adj_shadow_strength` also takes into account `shadow_strength`. I've removed this dependence because now with control over specular intensity, it would be very intransparent to modders that shadow strength should factor into reflection intensity. The same is true of the specular highlights themselves.